src/Hooks/FiltersTracking.md

```diff
@@ -40,33 +40,43 @@ This filter provides you with the ability to map you local storage tags got from
 **Filter example:**
 ```php
 // Map Hubspot fields with custom tags from local storage.
-\add_filter('es_forms_integration_hubspot_local_storage_map', [$this, 'getIntegrationHubspotLocalStorageMap'], 10, 2);
+\add_filter('es_forms_integration_hubspot_local_storage_map', [$this, 'getIntegrationHubspotLocalStorageMap'], 10, 3);
 
 /**
  * Map Hubspot fields with custom tags from local storage.
  *
- * @param array<int, mixed> $params Params from Hubspot integration.
+ * @param array<int, mixed> $params Params from Hubspot integration prepared for output.
  * @param array<int, mixed> $storage Data form storage.
+ * @param array<int, mixed> $originalParams Original params with all custom fields.
  *
  * @return array<int, mixed>
  */
-public function getIntegrationHubspotLocalStorageMap(array $params, array $storage): array
+public function getIntegrationHubspotLocalStorageMap(array $params, array $storage, array $originalParams): array
 {
 	if (!$storage) {
 		return $params;
 	}
 
-	foreach ($params as $key => $param) {
-		$name = $param['name'] ?? '';
+	// Additional tags to allow.
+	$allowedTags = \array_flip([
+		'utm_source',
+		'utm_content',
+		'utm_campaign',
+	]);
```

Comment:
Why not just write this as

```php
$allowedTags = [
	'utm_source' => 0,
	'utm_content' => 1,
	'utm_campaign' => 2,
];
```

I don't see this being dynamically updated, so you can save yourself from an unnecessary `array_flip` call.

Reply:
i don't know I like this approach better.

Reply:
Doing an `array_flip` and `foreach`-ing through it also seems more costly, especially as you can use this syntax without flipping with `in_array`, which should be a bit faster.

Reply:
> i don't know I like this approach better.

😂 

That's nice, but if you can do something a bit more performant then it should be done 😄 

I mean, I'm not sure what the intention is with that array, it could be that  

```php
$allowedTags = [
	'utm_source' => true,
	'utm_content' => true,
	'utm_campaign' => true,
];
```

Would be even better (and then checking for them using something like `isset`).

Reply:
👍